### PR TITLE
chore(ci): test dev shell building

### PR DESCRIPTION
**Pull request by dpc**

### CODE REVIEW

File: .github/workflows/ci-nix.yml [hunk 1]
```diff
@@ -67,6 +67,48 @@ jobs:
...
+      matrix:
+        host:
+          - linux
+          - macos
+        include:
+          - host: linux
+            runs-on: buildjet-4vcpu-ubuntu-2004
+            build-in-pr: true
+            timeout: 30
+            run-tests: true
```
Comment (bradleystachurski) [inline .github/workflows/ci-nix.yml:82]: [nit] `run-tests` isn't need in the dev shell check

File: .github/workflows/ci-nix.yml [hunk 2]
```diff
@@ -67,6 +67,48 @@ jobs:
...
+          nix_path: nixpkgs=channel:nixos-22.05
+      - uses: cachix/cachix-action@v14
+        if: github.event_name != 'pull_request' || matrix.build-in-pr
+        with:
+          name: fedimint
+          authToken: '${{ secrets.CACHIX_AUTH_TOKEN }}'
+        continue-on-error: true
+
+      - name: Build dev shell
+        if: github.event_name != 'pull_request' || matrix.build-in-pr
```
Comment (bradleystachurski) [inline .github/workflows/ci-nix.yml:109]: Looks like this takes ~1m in CI. Would it make sense to instead only run in the merge queue?

Comment (dpc) [reply .github/workflows/ci-nix.yml:107]: I already disabled macos one in PRs, but I guess I don't really care that much about Linux one as well - as long as `master` doesn't have a broken dev shell it's OK. And it barely ever happens anyway.

**Review by bradleystachurski — APPROVED** [review]

**Comment by bradleystachurski** [comment]
For future reference, this job should prevent these types of issues: https://github.com/fedimint/fedimint/issues/4217

**Review by oleonardolima — APPROVED** [review]
ACK a9dec97d69422519e24068a3de44148055d36af7

**Review by elsirion — APPROVED** [review]

**Comment by justinmoon** [comment]
Thanks @dpc!

**Pull request merged**